src/auto/dbus_interface_info.rs

```diff
@@ -0,0 +1,72 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use gio_sys;
+use glib;
+use glib::translate::*;
+use DBusMethodInfo;
+use DBusPropertyInfo;
+use DBusSignalInfo;
+
+glib_wrapper! {
+    #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+    pub struct DBusInterfaceInfo(Shared<gio_sys::GDBusInterfaceInfo>);
+
+    match fn {
+        ref => |ptr| gio_sys::g_dbus_interface_info_ref(ptr),
+        unref => |ptr| gio_sys::g_dbus_interface_info_unref(ptr),
+        get_type => || gio_sys::g_dbus_interface_info_get_type(),
+    }
+}
+
+impl DBusInterfaceInfo {
+    pub fn cache_build(&self) {
+        unsafe {
+            gio_sys::g_dbus_interface_info_cache_build(self.to_glib_none().0);
+        }
+    }
+
+    pub fn cache_release(&self) {
```

Comment:
`cache_build()` and `cache_release()` should probably get a return type that calls the latter on `Drop`

Reply:
It's a ref counted type of thing..

> If this has already been called with info , the existing cache is used and its use count is increased.

> Decrements the usage count for the cache for info built by g_dbus_interface_info_cache_build() (if any) and frees the resources used by the cache if the usage count drops to zero.

except an internal thing that you can't touch, so I don't think we need a return type

Reply:
The problem is that without having a such a type, you would "forget" to decrement it on panicking and probably also accidentally in functions with multiple return paths.

Each `cache_build()` would return a value (refcount +1), each `Drop` of them would call `cache_release() (refcount -1).

Am I missing something?

Reply:
Yeah I guess that's nice. I don't care about this API though, maybe I'll just ignore cache_ methods for now?

Reply:
Yes please, and afterwards we should create an issue with a list of all the missing things (like also the inspection and subtree APIs).